elasticsearch_collectd.py

```diff
@@ -993,10 +993,17 @@ def load_es_info(self):
         node_info = json['nodes'].itervalues().next()
         version = node_info['version']
         # a node is master eligible by default unless it's configured otherwise
+	# use a different json key for aws hosted elasticsearch
         master_eligible = True
-        if 'node' in node_info['settings'] and \
-           'master' in node_info['settings']['node']:
-            master_eligible = node_info['settings']['node']['master'] == 'true'
+	if 'settings' in node_info:
+            if 'node' in node_info['settings'] and \
+            'master' in node_info['settings']['node']:
+                master_eligible = node_info['settings']['node']['master'] == 'true'
+        elif 'roles' in node_info:
+            if 'master' in node_info['roles']:
+                master_eligible = True
+        else:
+            master_eligible = False
```

Comment:
Do you know of any circumstance when neither `settings` nor `roles` would be in `node_info`?  Since all nodes are master eligible by default, it seems like we shouldn't set it to false here unless it indicates a failed node or something.

Reply:
Hmmm good point. I'm not aware of any such circumstance but I haven't looked around much either. I'm not sure what GCP or Azure hosted ES might be doing instead, but maybe we cross that river when we get to it. I can amend out that line and have it marked as True/False similar to how its being done for the `settings` block.

Reply:
Yes, I would remove any defaulting to False.  Did you test this on both AWS and standard ES instances?

Reply:
not yet. I've yet to test on a standard ES instance. Will update with my findings.

Reply:
@harnitsignalfx @keitwb has there been any progress on this pr?